# PART II VISUALIZATION

## installing the necesaries packages

```{r}
if (!require("reshape2")) install.packages("reshape2")
library("reshape2")
if (!require("ggplot2")) install.packages("ggplot2")
library(ggplot2)
if (!require("plyr")) install.packages("plyr")
library(plyr)
```

# SNP PER CHROMOSOME
I look for the answer of *"what groups contributed more?"*

* creating a new data set, joining snp data to genotypes data
```{r}
joinedGenotype_SNP <- merge(snp_position, t(fang_et_al_genotypes), by.x = "SNP_ID", by.y = "row.names")
```

* enumerate chromosomes
```{r}
joinedGenotype_SNP$Chromosome<-factor(joinedGenotype_SNP$Chromosome,
                                      levels = c((1:10), "multiple", "unknown", "NA") )
```

* Count of SNP by Chromosome

```{r}
ggplot(joinedGenotype_SNP, aes(Chromosome)) + geom_bar()
```

## *what groups contributed more?"*

```{r}
ggplot(fang_et_al_genotypes, aes(Group)) +  geom_bar()
```
* the graph clearly shows that in order of importance ZMMLR, ZMPBA and ZMMIL groups contributed more 

## MISISNG DATA AND AMOUNT OF HETEROZYGOCITY

* I want just the genotypes, so I discard the three first informative columns

```{r}
genotypes<- colnames(fang_et_al_genotypes)[-c(1:3)]
```

* using melt function and rearrange by SNP's IDs

```{r}
genotypes.m <- melt(fang_et_al_genotypes, measure.vars = genotypes)
```

* recoding missing data ?/? as NA's

```{r}
genotypes.m[ genotypes.m == "?/?" ] = NA
```

* Creating a new column to indicate whether a particular site is homozygous

```{r}
genotypes.m$isHomozygous <- (genotypes.m$value=="A/A" | genotypes.m$value=="C/C" | genotypes.m$value=="G/G" | genotypes.m$value=="T/T")
```

* Sorting dataframe using Group and Species_ID values

```{r}
genotypes_ID <- genotypes.m[order(genotypes.m$Sample_ID),]
genotypes_Group <- genotypes.m[order(genotypes.m$Group),]
```


* Counting Homozygous and Heterozygous

```{r}
proportions_ID <- ddply(genotypes_ID, c("Sample_ID"), summarise, Homozygous=sum(isHomozygous, na.rm=TRUE), 
                     Heterozygous=sum(!isHomozygous, na.rm=TRUE), isNA=sum(is.na(isHomozygous)))
```


* using melt function 
```{r}
proportions_ID.m <- melt(proportions_ID, measure.vars = c("Homozygous", "Heterozygous", "isNA"))
```


* graph that shows the proportion of homozygous and heterozygous sites as well as missing data in each species 
```{r}
ggplot(proportions_ID.m,aes(x = Sample_ID, y= value, fill=variable)) + geom_bar(stat = "identity", position = "stack")
```

* The same process is made when the data is sorted for groups:

```{r}
proportions_Group <- ddply(genotypes_Group, c("Group"), summarise, Homozygous=sum(isHomozygous, na.rm=TRUE), Heterozygous=sum(!isHomozygous, na.rm=TRUE), isNA=sum(is.na(isHomozygous)))

proportions_Group.m <- melt(proportions_Group, measure.vars = c("Homozygous", "Heterozygous", "isNA"))
```

```{r}
ggplot(proportions_Group.m,aes(x = Group, y= value, fill=variable)) + geom_bar(stat = "identity", position = "stack")

```
* Finally I proposed a scaterplot to see the distribution of the SNP on the cromosomes 

```{r}
p <- ggplot(joinedGenotype_SNP, aes(x=SNP_ID, y=Chromosome)) + geom_point(size=1, color="black")

print(p)
```